### DISCUSSION TITLE
Connect to a Private Electrum Server

### DISCUSSION THREAD

**Discussion by STCB** [1200wd/bitcoinlib#401] [Q&A] [answered]
Hello !

I am new to using bitcoinlib, and just found out that I had to setup a Bcoin node to be my own RPC.
I find it kind of sad to duplicate my bitcoin node, as I already have a full bitcoin core setup and running...

So this situation lead me to wonder if a connexion to a private Electrum server could do the trick ?
To be honest I do not know a lot about electrum private servers, I just wonder if they give the ability to monitor UTXOs for "external" addresses and thus serve as an alternative for Bcoin in the future, as I think most people have an Electrum server connected to a bitcoin core, rather than Bcoin.

**Comment by mccwdev** [comment, marked as answer]
I assume it is possible to connect to a private electrum server, but there is no Service provider for it yet.
You can add one, but this requires some work and testing, you can find short manual here: https://bitcoinlib.readthedocs.io/en/latest/source/_static/manuals.add-provider.html

If you are not planning to write a Electrum service provider yourself you can add a feature request in the issues, otherwise please send a PR :wink:

In some specific situations you can also connect to a wallet in bitcoin core: https://coineva.com/bitcoinlib-and-multiple-bitcoin-core-wallets.html

**Reply by STCB** [reply]
Alright thanks !
I'll post a feature request then 👍
